data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+    fi
+}
+
+#Setup folder on host to be used as logs warehouse which will hold all host and guest logs at the last
+function setup_common_logs_folder() {
+        local logs_folder=$1
+        mkdir -p ${logs_folder}
+        chmod -R 777 ${logs_folder}
+	return $?
+}
+
+#Collect any desired logs from virtual machine by using virsh console and expect script
+function collect_logs_via_guest_console() {
+        local guest_domain=$1
+        local logs_folder=$2
+        shift
+        shift
+        local extra_logs=$@
+        local expfile="${logs_folder}/collect_logs_via_guest_console.exp"
+        local guest_transformed=${guest_domain//./_}
+        touch ${expfile}
+        chmod 777 ${expfile}
+        local hypervisor=`get_sles_hypervisor`
+
+cat <<EOF > ${expfile}
+set hypervisor [lindex \$argv 0]
+set guest_domain [lindex \$argv 1]
+set guest_transformed [lindex \$argv 2]
+set logs_folder [lindex \$argv 3]
+set extra_logs [lindex \$argv 4]
+set retry_times 3
+set ret_result 1
+
+while { \${retry_times} > 0 } {
+      if { \${hypervisor} == {KVM} } {
+         spawn virsh console --force \${guest_domain}
+      }
+      if { \${hypervisor} == {XEN} } {
+         spawn virsh console \${guest_domain}
+      }
+      set timeout 60
+      expect  {
+         -nocase "escape character" {send "\r\r"; exp_continue -continue_timer}
+         -nocase "password:" {send "\r"}
+         " #" {send "exit\r"}
+         -nocase "mistake|wrong|fault|error|fail|exception|not*found|timed*out" {puts "Can not establish virsh console to \${guest_domain}\r"; set ret_result 1}
+      }
+      expect {
+         -nocase "login: $"  {send "root\r"; set ret_result 0; exp_continue -continue_timer}
+         -nocase "password:" {send "novell\rcd ~\r"; exp_continue -continue_timer}
+         -nocase "mistake|wrong|fault|error|fail|exception|not*found|timed*out" {puts "Can not login virsh console to \${guest_domain}\r"; set ret_result 1}
+      }
+
+      if { \${ret_result} == 0 } {
+         set timeout 600
+         expect "~ #"
+         send "mkdir -p \${logs_folder};export time_stamp=\`date \'+%Y%m%d%H%M%S\'\`\r"
+         expect "~ #"
+         if { \${extra_logs} == {support_config} } {
+            send "rm -f -r \${logs_folder}/*supportconfig*\r"
+            send "supportconfig -t \${logs_folder} -B guest_\${guest_transformed}_supportconfig_\\\${time_stamp}\r"
+         }
+         if { \${extra_logs} != {support_config} && \${extra_logs} != "" } {
+            send "mkdir -p \${logs_folder}/guest_\${guest_transformed}_extra_logs;cp -r -f \${extra_logs} \${logs_folder}/guest_\${guest_transformed}_extra_logs\r"
+         }
+         expect {
+            "~ #" {send "export ret=\\\$?;cd /tmp;(if \[\[ \\\$ret -ne 0 ]];then printenv USER;fi)\r"; exp_continue -continue_timer}
+            "root" {puts "Can not collect \${extra_logs} from \${guest_domain}\r"; set ret_result 1;  exp_continue -continue_timer}
+            "tmp #" {send "cd ~\r"; send "exit\r"}
```

Comment:
Can you explain in what situation "root" and "tmp" are expected?

Reply:
expect script will continue to match "~#" constantly without doing anything else. So the only way to make it behave based test result is to change the prompt sign or print something else on the screen. In this case: FAIL-> printenv USER (root), then anyway expect "tmp #" to go ahead